lib/App/Switchman.pm

```diff
@@ -454,7 +490,19 @@ sub run
         }
     }
 
+    my $resources_wait_start_time = time;
+    local $SIG{ALRM} = sub {
+        # use 5 seconds gap to avoid the problem that the elapsed time can differ from the specified
+        if ($self->resources_wait_timeout && (time - $resources_wait_start_time) > ($self->resources_wait_timeout - 5)) {
```

Comment:
```perldoc -f alarm``` говорит, что время получения сигнала может быть на секунду меньше или больше указанного.
> (On some machines, unfortunately, the elapsed time may be up to one second less or more than you specified because of how seconds are counted,
               and process scheduling may delay the delivery of the signal even further.)